### DISCUSSION TITLE
Using the latest elastiknn with older version of ES?

### DISCUSSION THREAD

**Discussion by P3ngLiu** [alexklibisz/elastiknn#363] [Q&A] [answered]
@alexklibisz Hi, there! The latest verison of elastiknn is binned with ES 7.13.2, is there a way to use the latest elastiknn with older version of ES? Or can you give me some guidances to build a version on my own? Thanks!

**Comment by alexklibisz** [comment, marked as answer]
Hi - unfortunately there's no way to (easily) backport it, as the Elasticsearch JVM code is not backwards compatible across releases. If you really cannot update Elasticsearch, here are two suggestions:
1. The project has been relatively stable for most of 2021. Most new releases are just to keep up with new Elasticsearch versions. It's likely there's an older version that should work quite well. So I suggest reading the release notes to see if there's actually any meaningful change.
2. Fork the project and build/release an internal/personal copy for your current version of Elasticsearch. You can look at old PRs to see how to change the version and how the API has changed from version to version.
